src/lib.rs

```diff
@@ -279,6 +282,12 @@ impl UnixStream {
         self.io.get_ref().peer_addr()
     }
 
+    /// Returns effective credentials of the process which called `connect` or `socketpair`.
+    #[cfg(any(target_os = "linux", target_os = "macos"))]
+    pub fn peer_cred(&self) -> io::Result<UCred> {
```

Comment:
Do you think all other unices have an implementation of this function? If that's the case then this should just remove the `#[cfg]`.